### ISSUE TITLE
The default PointRend code with cityscapes semantic segmentation runs into NaN

### ISSUE THREAD

**Issue by liuzhuang13** [facebookresearch/detectron2#1455]
Thanks for the high-quality code. I tried three times to run the pointrend default yaml config for cityscape semantic segmentation, but it runs into loss being NaN in anywhere between tens iterations and thousands of iterations.

## Instructions To Reproduce the Issue:

1. what changes you made (`git diff`) or what code you wrote
```
No code was changed
```
2. what exact command you run:
in the pointrend project:
python train_net.py --config-file configs/SemanticSegmentation/pointrend_semantic_R_101_FPN_1x_cityscapes.yaml --num-gpus 8

3. what you observed (including __full logs__):
```
[05/18 20:20:33 d2.utils.events]:  eta: 15:20:48  iter: 2459  total_loss: 0.865  loss_sem_seg: 0.265  loss_sem_seg_point: 0.605  time: 1.8540  data_time: 1.3640  lr: 0.010000  max_mem: 4673M
[05/18 20:21:01 d2.utils.events]:  eta: 15:19:10  iter: 2479  total_loss: 0.854  loss_sem_seg: 0.257  loss_sem_seg_point: 0.606  time: 1.8504  data_time: 0.6700  lr: 0.010000  max_mem: 4673M
[05/18 20:21:37 d2.utils.events]:  eta: 15:17:54  iter: 2499  total_loss: 0.858  loss_sem_seg: 0.258  loss_sem_seg_point: 0.607  time: 1.8498  data_time: 1.1570  lr: 0.010000  max_mem: 4673M
ERROR [05/18 20:21:56 d2.engine.train_loop]: Exception during training:
Traceback (most recent call last):
  File "/home/zhuangl/pointrend/detectron2/detectron2/engine/train_loop.py", line 132, in train
    self.run_step()
  File "/home/zhuangl/pointrend/detectron2/detectron2/engine/train_loop.py", line 217, in run_step
    self._detect_anomaly(losses, loss_dict)
  File "/home/zhuangl/pointrend/detectron2/detectron2/engine/train_loop.py", line 238, in _detect_anomaly
    raise FloatingPointError(
FloatingPointError: Loss became infinite or NaN at iteration=2512!
loss_dict = {'loss_sem_seg': tensor(nan, device='cuda:0', grad_fn=<MulBackward0>), 'loss_sem_seg_point': tensor(nan, device='cuda:0', grad_fn=<NllLoss2DBackward>)}
[05/18 20:21:56 d2.engine.hooks]: Overall training speed: 2510 iterations in 1:17:20 (1.8487 s / it)
[05/18 20:21:56 d2.engine.hooks]: Total training time: 1:17:25 (0:00:05 on hooks)
Traceback (most recent call last):
  File "train_net.py", line 126, in <module>
    launch(
  File "/home/zhuangl/pointrend/detectron2/detectron2/engine/launch.py", line 50, in launch
    mp.spawn(
  File "/home/zhuangl/anaconda3/envs/prend/lib/python3.8/site-packages/torch/multiprocessing/spawn.py", line 200, in spawn
    return start_processes(fn, args, nprocs, join, daemon, start_method='spawn')
  File "/home/zhuangl/anaconda3/envs/prend/lib/python3.8/site-packages/torch/multiprocessing/spawn.py", line 158, in start_processes
    while not context.join():
  File "/home/zhuangl/anaconda3/envs/prend/lib/python3.8/site-packages/torch/multiprocessing/spawn.py", line 119, in join
    raise Exception(msg)
Exception:

-- Process 7 terminated with the following error:
Traceback (most recent call last):
  File "/home/zhuangl/anaconda3/envs/prend/lib/python3.8/site-packages/torch/multiprocessing/spawn.py", line 20, in _wrap
    fn(i, *args)
  File "/home/zhuangl/pointrend/detectron2/detectron2/engine/launch.py", line 89, in _distributed_worker
    main_func(*args)
  File "/home/zhuangl/pointrend/detectron2/projects/PointRend/train_net.py", line 120, in main
    return trainer.train()
  File "/home/zhuangl/pointrend/detectron2/detectron2/engine/defaults.py", line 401, in train
    super().train(self.start_iter, self.max_iter)
  File "/home/zhuangl/pointrend/detectron2/detectron2/engine/train_loop.py", line 132, in train
    self.run_step()
  File "/home/zhuangl/pointrend/detectron2/detectron2/engine/train_loop.py", line 217, in run_step
    self._detect_anomaly(losses, loss_dict)
  File "/home/zhuangl/pointrend/detectron2/detectron2/engine/train_loop.py", line 238, in _detect_anomaly
    raise FloatingPointError(
FloatingPointError: Loss became infinite or NaN at iteration=2512!
loss_dict = {'loss_sem_seg': tensor(nan, device='cuda:7', grad_fn=<MulBackward0>), 'loss_sem_seg_point': tensor(nan, device='cuda:7', grad_fn=<NllLoss2DBackward>)}
```
4. please simplify the steps as much as possible so they do not require additional resources to
	 run, such as a private dataset.

## Expected behavior:

The training should proceed without NaN error.

## Environment:

Provide your environment information using the following command:
wget -nc -q https://github.com/facebookresearch/detectron2/raw/master/detectron2/utils/collect_env.py && python collect_env.py

sys.platform              linux
Python                    3.8.2 (default, May  7 2020, 20:00:49) [GCC 7.3.0]
numpy                     1.18.4
detectron2                0.1.3 @/home/zhuangl/pointrend/detectron2/detectron2
detectron2 compiler       GCC 7.4
detectron2 CUDA compiler  10.1
detectron2 arch flags     sm_61
DETECTRON2_ENV_MODULE     <not set>
PyTorch                   1.5.0+cu101 @/home/zhuangl/anaconda3/envs/prend/lib/python3.8/site-packages/torch
PyTorch debug build       False
CUDA available            True
GPU 0,1,2,3,4,5,6,7,8,9   TITAN Xp
CUDA_HOME                 /usr/local/cuda
NVCC                      Cuda compilation tools, release 10.1, V10.1.168
Pillow                    7.1.2
torchvision               0.6.0+cu101 @/home/zhuangl/anaconda3/envs/prend/lib/python3.8/site-packages/torchvision
torchvision arch flags    sm_35, sm_50, sm_60, sm_70, sm_75
fvcore                    0.1.1.post200513
cv2                       4.2.0
------------------------  ---------------------------------------------------------------------------------------
PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2019.0.5 Product Build 20190808 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v0.21.1 (Git Hash 7d2fd500bc78936d1d648ca713b901012f470dbc)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 10.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_37,code=compute_37
  - CuDNN 7.6.3
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_INTERNAL_THREADPOOL_IMPL -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF, 

Thanks in advance

**Comment by ppwwyyxx** [comment]
Hi, we observed something similar with this config but it is able to work after a few retries. We're looking at ways to make it train more stable at the beginning.

**Comment by hsfzxjy** [comment]
@ppwwyyxx  I encountered another strange error after training PointRend on Cityscapes Semantic Segmentation for  around 600 epochs. The error message is:
```
Exception during training:
Traceback (most recent call last):
  File "/msravcshare/v-jinxi/pointrend/detectron2/engine/train_loop.py", line 132, in train
    self.run_step()
  File "/msravcshare/v-jinxi/pointrend/detectron2/engine/train_loop.py", line 217, in run_step
    self._detect_anomaly(losses, loss_dict)
  File "/msravcshare/v-jinxi/pointrend/detectron2/engine/train_loop.py", line 237, in _detect_anomaly
    if not torch.isfinite(losses).all():
RuntimeError: CUDA error: device-side assert triggered
[05/21 10:28:25 d2.engine.hooks]: Overall training speed: 634 iterations in 0:10:56 (1.0353 s / it)
[05/21 10:28:25 d2.engine.hooks]: Total training time: 0:10:57 (0:00:01 on hooks)` failed.
/opt/conda/conda-bld/pytorch_1579027003190/work/aten/src/ATen/native/cuda/IndexKernel.cu/opt/conda/conda-bld/pytorch_1579027003190/work/aten/src/ATen/native/cuda/IndexKernel.cu

/opt/conda/conda-bld/pytorch_1579027003190/work/aten/src/ATen/native/cuda/IndexKernel.cu:60: lambda [](int)->auto::operator()(int)->auto: block: [22,0,0], thread: [0,0,0] Assertion `index >= -sizes[i] && index < sizes[i] && "index out of bounds"` failed.
<similar messages repeating ...>
```

Any suggestions for this error?

**Comment by bluesky314** [comment]
@ppwwyyxx Any update on this?

**Comment by ppwwyyxx** [comment]
fixed in 81d5a87763bfc71a492b5be89b74179bd7492f6b
